### PR TITLE
test: coverage report to exclude test/util and to not run race detection

### PR DESCRIPTION
**Pull request by cmwaters**

### CODE REVIEW

**Review by rootulp — APPROVED** [review]
[non-blocking][question] why did we ignore `/malicious` from test coverage in https://github.com/celestiaorg/celestia-app/pull/2008 ?

I don't see `/malicious` ignored from the test commands in the Makefile: https://github.com/celestiaorg/celestia-app/blob/main/Makefile so it seems safe to run test on it + collect code coverage for it

**Comment by cmwaters** [comment]
> I don't see `/malicious` ignored from the test commands in the Makefile: https://github.com/celestiaorg/celestia-app/blob/main/Makefile so it seems safe to run test on it + collect code coverage for it

Yeah I'm indifferent about this. I'm not too sure who's using the code coverage data at all. You could also argue that code coverage for parts that can't possibly be hit in production (i.e. test infrastructure) isn't important to publish.

**Review by rootulp — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]
I guess the `e2e` framework and `txsim` also could be excluded as they're part of the infrastructure. Or could be moved under `util` too.

File: scripts/test_cover.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 set -e
 
 # Define the directories to exclude
-EXCLUDE_DIRS=("/malicious")
+EXCLUDE_DIRS=("/test/util")
```
Comment (staheri14) [inline scripts/test_cover.sh:5]: A suggestion about the PR title: This PR does more than "removing race detection", it also "excludes the "/test/util" directory from code coverage analysis".  You may consider updating the title.

**Review by staheri14 — COMMENTED** [review]
It LGTM, however, I have a question: Am I correct in assuming that this script is exclusively for generating a test coverage report? In other words, do we still execute the tests located in the test/util directory, such as [this test](https://github.com/celestiaorg/celestia-app/blob/4725fd8925288fc03b42b1180526e07eaa2b927b/test/util/blobfactory/payforblob_factory_test.go#L15), or not?

**Comment by SweeXordious** [comment]
> It LGTM, however, I have a question: Am I correct in assuming that this script is exclusively for generating a test coverage report? In other words, do we still execute the tests located in the test/util directory, such as [this test](https://github.com/celestiaorg/celestia-app/blob/4725fd8925288fc03b42b1180526e07eaa2b927b/test/util/blobfactory/payforblob_factory_test.go#L15), or not?

The tests are run in here: 
https://github.com/celestiaorg/celestia-app/blob/81ee99bf6a805cad7d3a20c544147c27c1b0aa82/.github/workflows/test.yml#L18-L19

What this PR touches is this, to generate the report:
https://github.com/celestiaorg/celestia-app/blob/81ee99bf6a805cad7d3a20c544147c27c1b0aa82/.github/workflows/test.yml#L21-L36

But as you said, it would be better if the title of the PR is adapted to also refer to the EXCLUDE directories change.

**Review by staheri14 — APPROVED** [review]
LGTM! (my suggestion for a change in the PR title still holds :) )

**Comment by staheri14** [comment]
Thanks @SweeXordious for clarification on the tests 🙏

**Pull request merged**